### PR TITLE
181563755 user page

### PR DESCRIPTION
**Pull request by ChuanHuoGe**
Abstract footnote into a global partial. 
Make main_content(previous book_content), topbar, table_of_contents global partials. You could review views/application/main_content.html.erb first
Add user page(views/users/_user_page.html.erb) and use main_content as the template.

### CODE REVIEW

**Comment by whellen0112** [comment]
Checking the herokuapp at https://esaas169l.herokuapp.com/, all the buttons work well and login button work successfully. But when I click the user page button at the top right corner, I can't access the user page cuz it doesn't exist. Is everything pushed to heroku?

**Comment by ChuanHuoGe** [comment]
> Checking the herokuapp at https://esaas169l.herokuapp.com/, all the buttons work well and login button work successfully. But when I click the user page button at the top right corner, I can't access the user page cuz it doesn't exist. Is everything pushed to heroku?

No I didn't, but I just pushed to heroku

**Comment by whellen0112** [comment]
> > Checking the herokuapp at https://esaas169l.herokuapp.com/, all the buttons work well and login button work successfully. But when I click the user page button at the top right corner, I can't access the user page cuz it doesn't exist. Is everything pushed to heroku?
> 
> No I didn't, but I just pushed to heroku

Is the url still same? https://esaas169l.herokuapp.com/ is not working somehow.

File: app/views/users/show.html.erb [hunk 1]
```diff
@@ -0,0 +1,96 @@
+<!DOCTYPE html>
+<!-- This is coming from book_content -->
+<html>
+    <head>
+        <meta charset="utf-8" />
+        <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+        <title>1. Introduction to Software as a Service, Agile Development, and Cloud Computing &#8212; saasbook  documentation</title>
```
Comment (JDDelrosario) [inline app/views/users/show.html.erb]: Remove this line

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        <%= stylesheet_link_tag "all.min.css" %>
+        <%= stylesheet_link_tag "pygments.css" %>
+        <%= stylesheet_link_tag "sphinx-book-theme.acff12b8f9c144ce68a297486a2fa670.css" %>
+
+        
+        <%= javascript_include_tag "vendor.js" %>
+        <%= javascript_include_tag "custom.js" %>
+        <% if user_signed_in? && @section_id != -1 %>
+            <%= javascript_include_tag "highlight-annotate.js" %>
+        <% end %>
```
Comment (JDDelrosario) [inline app/views/users/show.html.erb]: Remove these lines

**Review by JDDelrosario — APPROVED** [review]

File: app/views/saasbook/_chapter11section4.html.erb [hunk 3]
```diff
@@ -78,145 +77,3 @@ all variants; hence it is usually implemented using inheritance. With Strategy,
 task is the same, but the set of steps may be different in each variant; hence it is usually
 implemented using composition. Figure 11.12 shows how either pattern could be applied to the
 report formatter. If every kind of formatter followed the same high- level steps—for example,
-generate the header, generate the report body, and then generate the footer—we could use
```
Comment (Lucitic) [inline app/views/saasbook/_chapter11section4.html.erb]: i dont think any of this is supposed to be removed

File: app/views/saasbook/_chapter5section1.html.erb [hunk 4]
```diff
@@ -41,195 +40,3 @@ Figure 5.1 shows.</p>
 
 <p>Partials rely heavily on convention over configuration. Their names must begin with an underscore
 (we used <code class="code docutils literal notranslate"><span class="pre">_movie_form.html.erb</span></code>) which is <em>absent from</em> the code that references the partial. A partial may
-be in a different directory as the view that uses it, in which case a path such as <strong>’layouts/footer’</strong> would
```
Comment (Lucitic) [inline app/views/saasbook/_chapter5section1.html.erb]: this shouldnt be deleted

File: app/views/saasbook/_chapter6section5.html.erb [hunk 5]
```diff
@@ -33,98 +32,3 @@ quickly jump between sections:</p>
 <li><p><code class="code docutils literal notranslate"><span class="pre">h1...h6</span></code> are elements that give your page an outline. Most pages start with <code class="code docutils literal notranslate"><span class="pre">h1</span></code>.</p></li>
 <li><p><code class="code docutils literal notranslate"><span class="pre">nav</span></code> is used for sections of links that direct users to different parts of your website. Commonly, this is used in a “navigation bar” at the top of a web page.</p></li>
 <li><p><code class="code docutils literal notranslate"><span class="pre">main</span></code> This is for the body of your web page. There should only be one of these per page.</p></li>
-<li><p><code class="code docutils literal notranslate"><span class="pre">header</span></code> and <code class="code docutils literal notranslate"><span class="pre">footer</span></code> These are common sections that you might include. They should be <em>outside</em> of the main content.</p></li>
```
Comment (Lucitic) [inline app/views/saasbook/_chapter6section5.html.erb]: ditto

**Review by Lucitic — CHANGES_REQUESTED** [review]

**Review by JDDelrosario — APPROVED** [review]

**Review by Lucitic — APPROVED** [review]

**Pull request merged**